### DISCUSSION TITLE
'NoneType' object has no attribute 'find' Error?

### DISCUSSION THREAD

**Discussion by aloeee3** [nianeyna/ao3downloader#107] [Q&A]
Hello,

The Ao3 Downloader is great! I love it. I have been using it fine so far but have hit this error "'NoneType' object has no attribute 'find'" today when trying to download a tag. The tag got downloaded up to 4 pages before stopping with this error and the page itself has a lot of exclusions on it since I try to filter out as much as possible to get to what I want. 

I tried to look this error up but wasn't able to find anything concrete. I am not in any way a coder so I would love some advice. My python version is 3.10.8 if this helps.

This program is amazing! I am using it more or less every day and am really building my library of fics and I would love some advice on how to fix this.

Thank you!

**Comment by nianeyna** [comment]
Hi! Thanks for reporting this. I am a little behind on ao3downloader support (work has been crazy) but I'll try to take a look at this soon. If you're comfortable, please email me (nianeyna at gmail) your log.jsonl file so I can do some digging. Most likely someone put some odd markup in their summary that the code doesn't know how to handle.

**Reply by aloeee3** [reply]
Thank you so much! I did as you suggested. Thank you very much for all you do with this program!

**Reply by alexandra-e-ressler** [reply]
Hello, I'm running into a similar error.  When it asks me if I want to log in, clicking y returns 

'NoneType' object has no attribute 'find'
'd' to display the menu again

I've also tried downloaging without logging in, and while it says

generating list of work links that are already in the downloads folder (will be skipped)
downloading works

I can't find any files getting downloaded anywhere.  I've checked the downloads folder I used previously and it just has all the old ones.  I don't know if these are related, but if you have any ideas, I'd really appreciate it!

**Reply by gideonav** [reply]
Hi! I'm having the same issue as well right now. I know you're probably busy, but if you have time to get to it this downloader has been such a lifesaver for me!

**Reply by nianeyna** [reply]
@alexandra-e-ressler @gideonav please make sure you're using the latest version of ao3downloader (aka redownload it from the repository, and use that version). If the issue persists after that let me know.

**Reply by knittingowl** [reply]
I just updated to the latest version and am having the same errors. Thanks for everything!

**Reply by beebutler** [reply]
Lastest version here with same issue as well.

**Reply by beebutler** [reply]
We're back to logging in and downloading.

**Comment by nianeyna** [comment]
Since this keeps coming up I should probably state for the record here that `'NoneType' object has no attribute 'find'` is just the generic error that python spits out when it encounters something it can't read. To my knowledge, there are _at least_ three entirely distinct issues reported in this thread. That's not to say you shouldn't continue reporting such errors of course, just don't assume that the fix will be the same each time (for example, in one case here the issue was that I needed to update the script, while in the most recent case, the issue was that ao3 was temporarily blocking bot traffic and the fix was just to wait a bit).
